File: CS/ServerParams/ProductReport.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.Parameters;

namespace ServerParams
{
    public partial class ProductReport : DevExpress.XtraReports.UI.XtraReport
    {
        public ProductReport()
        {
            InitializeComponent();
        }

        private void ProductReport_ParametersRequestSubmit(object sender, ParametersRequestEventArgs e)
        {
            if (Parameters["CategoryID"] != null)
                productsTableAdapter.FillByCategoryID(northwind1.Products, (int)Parameters["CategoryID"].Value);
        }

    }
}

File: VB/ServerParams/ProductReport.vb
Imports System.Drawing
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.Parameters

Namespace ServerParams

    Public Partial Class ProductReport
        Inherits DevExpress.XtraReports.UI.XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub ProductReport_ParametersRequestSubmit(ByVal sender As Object, ByVal e As ParametersRequestEventArgs)
            If Parameters("CategoryID") IsNot Nothing Then productsTableAdapter.FillByCategoryID(northwind1.Products, CInt(Parameters(CStr("CategoryID")).Value))
        End Sub
    End Class
End Namespace
